src/test/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilderTest.java

```diff
@@ -35,17 +37,32 @@ class CriteriaQueryBuilderTest {
 
   private static final SnapshotBuilderSettings SNAPSHOT_BUILDER_SETTINGS =
       new SnapshotBuilderSettings()
-          .domainOptions(List.of(new SnapshotBuilderDomainOption().id(19)))
+          .domainOptions(
+              List.of(
+                  (SnapshotBuilderDomainOption)
+                      new SnapshotBuilderDomainOption()
+                          .kind(SnapshotBuilderOption.KindEnum.DOMAIN)
+                          .tableName("condition_occurrence")
+                          .columnName("condition_concept_id")
+                          .id(19),
+                  (SnapshotBuilderDomainOption)
+                      new SnapshotBuilderDomainOption()
+                          .kind(SnapshotBuilderOption.KindEnum.DOMAIN)
+                          .tableName("drug_exposure")
+                          .columnName("drug_concept_id")
+                          .id(13)))
           .programDataOptions(
               List.of(
-                  new SnapshotBuilderProgramDataOption()
-                      .kind(SnapshotBuilderProgramDataOption.KindEnum.LIST)
-                      .columnName("list_column_name")
-                      .id(1),
-                  new SnapshotBuilderProgramDataOption()
-                      .kind(SnapshotBuilderProgramDataOption.KindEnum.RANGE)
-                      .columnName("range_column_name")
-                      .id(0)));
+                  (SnapshotBuilderProgramDataListOption)
+                      new SnapshotBuilderProgramDataListOption()
+                          .columnName("list_column_name")
```

Comment:
I am a little confused on the purpose of the tableName and columnName fields. I know this is just test data but does columnName have any meaning/purpose without tableName?

Reply:
That makes sense. So then if I understand, do you want to change the yaml file to require these fields (tableName and columnName)? and then add tableName to these ProgramDataOptions?

Reply:
IMO this is one of the problems of fluent-style constructors, it makes it hard for a class to ensure that all required fields are set, unless it's really a builder and you have a `build()` step. Although this is all test data so it doesn't really matter too much.

It might end up looking just as messy, but another approach is to use JSON literals (which you can pass as an argument to `formatted()`) and then using `ObjectMapper` to read the JSON as a java object. I _think_ it will honor `required = true` in `@Schema` but am not completely sure.

Reply:
I would prefer to avoid JSON literals - I find them harder to follow/easier to break than staying within a Java context.